tests/tel2puml/find_unique_graphs/test_otel_data_holder.py

```diff
@@ -222,61 +240,63 @@ def test_commit_batched_data_failure(
     ) -> None:
         """Tests commit_batched_data_to_database method, with failure."""
 
-        holder = SQLDataHolder(mock_sql_config)
-
-        # Test the IntegrityError case
-        with patch.object(
-            holder,
-            "batch_insert_node_models",
-            side_effect=IntegrityError(
-                "IntegrityError", None, Exception("IntegrityError")
-            ),
-        ), patch.object(holder.session, "rollback") as mock_rollback:
-            with pytest.raises(Exception) as context:
-                holder.commit_batched_data_to_database()
-            mock_rollback.assert_called_once()
-            assert "IntegrityError" in str(context.value)
-
-        # Test the OperationalError case
-        with patch.object(
-            holder,
-            "batch_insert_node_models",
-            side_effect=OperationalError(
-                "OperationalError", None, Exception("OperationalError")
-            ),
-        ), patch.object(holder.session, "rollback") as mock_rollback:
-            with pytest.raises(Exception) as context:
-                holder.commit_batched_data_to_database()
-            mock_rollback.assert_called_once()
-            assert "OperationalError" in str(context.value)
-
-        # Test the unexpected error case
-        with patch.object(
-            holder,
-            "batch_insert_node_models",
-            side_effect=Exception("Unexpected Error"),
-        ), patch.object(holder.session, "rollback") as mock_rollback:
-            with pytest.raises(Exception) as context:
-                holder.commit_batched_data_to_database()
-            mock_rollback.assert_called_once()
-            assert "Unexpected Error" in str(context.value)
+        with SQLDataHolder(mock_sql_config) as holder:
```

Comment:
again can be removed re previous comments

Reply:
Do we need it in this case, since we are patching `patch.object(holder.session, "rollback") as mock_rollback:` which requires the session, which will come from the context manager?